### DISCUSSION TITLE
Static objects collidable pairs, how?

### DISCUSSION THREAD

**Discussion by DynaBimm** [bepu/bepuphysics2#290] [Q&A]
Hello!

I need to get collidable pairs of AABBs and static geometry with no dynamic objects in scene. Also i need the most precise solution, maybe in a cost of speed. So, maybe there's some tricks?

Comment my steps, please.
1. BroadPhase, create tree structure of Bounding Boxes.
2. "GetOverlaps" of bounding boxes. Or should i use "BoxPairTester"?
3. Use info from step 2 and get more precise results for collidable pairs of meshes. Also, what if mesh is concave?

Some info about meshes. It's rather simple shapes. Mostly boxes, or boxes with rectagular cuts inside or cuts placed on edges (walls with windows and doors). Maybe i could do some tests to represent complex meshes with simple boxes to get collisons? And if simple representaion test fails, i'll use slow mesh collision test& 

Sorry fo dumb questions, i'm totally new in physics engines, so i'm trying not to make totally stupid things.
Thanks a lot for BepuPhysics!

**Comment by RossNordby** [comment]
> 1. BroadPhase, create tree structure of Bounding Boxes.

Yup. If you're using the simulation's broad phase, there's no need to create another tree, you can just use it.

> 2. "GetOverlaps" of bounding boxes. Or should i use "BoxPairTester"?

Yup, `GetOverlaps`. `BoxPairTester` is for narrow phase testing between two `Box` shapes.

> 3. Use info from step 2 and get more precise results for collidable pairs of meshes. Also, what if mesh is concave?

Yup. The mesh being concave doesn't matter; the `Mesh` shape supports arbitrary triangle soups. (The triangles *are* one sided, though.)

Check the [CollisionQueryDemo](https://github.com/bepu/bepuphysics2/blob/master/Demos/Demos/CollisionQueryDemo.cs) for an example.
